### DISCUSSION TITLE
New column indicating the total time needed to obtain the result of experiment X.

### DISCUSSION THREAD

**Discussion by raspberryananas** [microsoft/nni#3773] [New Feature Design Discussion]
**What would you like to be added**:

New column indicating the total time needed to obtain the result of experiment X.

**Why is this needed**:

The user can see at a glance how much time it took for experiment X to produce its result.

**Without this feature, how does current nni work**：

Until now, the times have to be summed up in a time-consuming way.

In the example below I'm only interested in how much time it took to get the best configuration (in this case the lowest `val_loss` in experiment 6). Add time for experiment 1+2+3+4+5+6

![image](https://user-images.githubusercontent.com/12211578/63407231-493ddd00-c3ec-11e9-8636-1d831ffb0d15.png)

**Comment by QuanluZhang** [comment]
Thanks @martsalz , I can understand how you calculate this number for trial 6. As you want a column, I am wondering how to calculate this number for trial 13 (or other trials) who is not the best one.

And is this number related to trial concurrency? How to calculate this number if `trialConcurrency` is larger than 1, for example 2.

**Comment by raspberryananas** [comment]
We assume that `trialConcurrency = 1`

**Comment by scarlett2018** [comment]
@martsalz - does **trial ending time** work for this scenario? in the home page, we have the experiment starting time. We can easily add the trial starting and ending time columns, we already have that in the API.

**Comment by raspberryananas** [comment]
> does trial ending time work for this scenario?

@scarlett2018  What exactly do you mean?

I think it would be useful to add a column for trial starting and ending time.

@QuanluZhang It would also satisfy if only the total time required for the best experiment is specified. I would like to see at a glance how much time has passed between starting the experiment and getting the best configuration - without manual calculating.

**Comment by scarlett2018** [comment]
> > does trial ending time work for this scenario?
> 
> @scarlett2018 What exactly do you mean?
> 
> I think it would be useful to add a column for trial starting and ending time.

@martsalz - I'm thinking about how generic is it to add a "total time" column. The trial starting and ending time are trial information we already have but hidden from UI, we can easily add them to the UI. But when adding a "total time", I'm a little concern about whether it will confuse users who does not have needs like your scenario.

**Comment by raspberryananas** [comment]
In AutoML, in particular the hyperparameter tuning, the essential goal is to automatically find a good configuration - in a short time. Therefore, as a user, I'm only interested in how time-consuming the search took - a matter that is important and interesting for everyone.

**Comment by pkuyym** [comment]
@martsalz The total time seems not that important when we search for the 'best' hyper-parameter since we never know which setting is the final best one. Even we get the time cost for the best setting, this time is not reliable since the reproduction is difficult.
